graph/SchemaGraph.java

```diff
@@ -367,8 +367,8 @@ public TypeVertex update(TypeVertex vertex, String oldLabel, @Nullable String ol
         final String oldScopedLabel = scopedLabel(oldLabel, oldScope);
         final String newScopedLabel = scopedLabel(newLabel, newScope);
         try {
-            multiLabelLock.lockWrite();
             final TypeVertex type = getType(newLabel, newScope);
+            multiLabelLock.lockWrite();
```

Comment:
reordered to prevent deadlock: `getType()` acquires the `multiLabelLock.readLock()`